_pytest/recwarn.py

```diff
@@ -28,9 +28,18 @@ def pytest_namespace():
             'warns': warns}
 
 
-def deprecated_call(func, *args, **kwargs):
+def deprecated_call(func=None, *args, **kwargs):
     """Assert that ``func(*args, **kwargs)`` triggers a DeprecationWarning.
+
+    This function can be used as a context manager::
+
+        >>> with deprecated_call():
+        ...    myobject.deprecated_method()
     """
+    if not func:
+        warnings.simplefilter('always')
```

Comment:
I believe this is because in a list of filters - the first one that matches will apply, and upon inspecting the filters (before the `always` is appended):
```
[('ignore', None, <type 'exceptions.DeprecationWarning'>, None, 0), 
('ignore', None, <type 'exceptions.PendingDeprecationWarning'>, None, 0),
 ('ignore', None, <type 'exceptions.ImportWarning'>, None, 0),
('ignore', None, <type 'exceptions.BytesWarning'>, None, 0)]
```
I see now why appending `always` will never work. The question is : is there a point in appending this filter in the first place(instead of prepending) since it is possible to have one that will match before it.